tutorials/distributed-ml/torch-tutorial-GAN/train.py

```diff
@@ -0,0 +1,471 @@
+        )
+
+        self.save_plots_and_images(epoch)
+
+    def validation_epoch(self, epoch: int):
+        g_validation_losses = []
+        g_validation_accuracy = []
+        self.discriminator.eval()
+        self.generator.eval()
+        validation_loss = torch.tensor(0.0, device=self.device)
+        for batch_idx, (real_images, _) in enumerate(
+                self.validation_dataloader):
+            loss_gen, accuracy_gen = self.validation_step(
+                real_images, batch_idx)
+            g_validation_losses.append(loss_gen)
+            g_validation_accuracy.append(accuracy_gen)
+            self.validation_glob_step += 1
+
+        # Aggregate and log metrics
+        validation_loss = torch.mean(torch.stack(g_validation_losses))
+        self.log(
+            item=validation_loss.item(),
+            identifier='gen_validation_loss_epoch',
+            kind='metric',
+            step=self.validation_glob_step,
+        )
+        validation_accuracy = torch.mean(torch.stack(g_validation_accuracy))
+        self.log(
+            item=validation_accuracy.item(),
+            identifier='gen_validation_accuracy_epoch',
+            kind='metric',
+            step=self.validation_glob_step,
+        )
+
+        return validation_loss
+
+    def train_step(self, real_images, batch_idx):
+        real_images = real_images.to(self.device)
+        batch_size = real_images.size(0)
+        real_labels = torch.full(
+            (batch_size,), REAL_LABEL,
+            dtype=torch.float, device=self.device)
+        fake_labels = torch.full(
+            (batch_size,), FAKE_LABEL,
+            dtype=torch.float, device=self.device)
+
+        # Train Discriminator with real images
+        output_real = self.discriminator(real_images)
+        lossD_real = self.criterion(output_real, real_labels)
+        # Generate fake images and train Discriminator
+        noise = torch.randn(batch_size, NOISE_DIM, 1, 1, device=self.device)
+
+        fake_images = self.generator(noise)
+        output_fake = self.discriminator(fake_images.detach())
+        lossD_fake = self.criterion(output_fake, fake_labels)
+
+        loss = (lossD_real+lossD_fake)/2
+
+        self.optimizerD.zero_grad()
+        loss.backward()
+        self.optimizerD.step()
+
+        # Train Generator
+        output_fake = self.discriminator(fake_images)
+        lossG = self.criterion(output_fake, real_labels)
+        self.optimizerG.zero_grad()
+        lossG.backward()
+        self.optimizerG.step()
+
+        return lossG, loss
+
+    def validation_step(self, real_images, batch_idx):
+        real_images = real_images.to(self.device)
+        batch_size = real_images.size(0)
+        real_labels = torch.full((batch_size,), REAL_LABEL,
+                                 dtype=torch.float, device=self.device)
+        fake_labels = torch.full((batch_size,), FAKE_LABEL,
+                                 dtype=torch.float, device=self.device)
```

Comment:
use torch zeros and ones, and remove `REAL_LABEL` and `FAKE_LABEL` from this file.